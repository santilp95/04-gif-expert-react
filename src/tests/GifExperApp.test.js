import'@testing-library/jest-dom';
import React from 'react';
const { shallow } = require("enzyme");
const { GifExperApp } = require("../GifExperApp");

describe('Pruebas en <GifExperApp/>', () => {

    test('debe funcionar correctamente', () => {
        const wrapper = shallow(<GifExperApp/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar una lista de categorias', () => {
        const categories = ['Naruto','dragon ball'];
        const wrapper = shallow(<GifExperApp defalutCategories={categories}/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });
    
    
});
