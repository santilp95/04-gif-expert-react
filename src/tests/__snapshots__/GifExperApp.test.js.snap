// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExperApp/> debe funcionar correctamente 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GifExperApp/> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExperApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Naruto"
      key="Naruto"
    />
    <GifGrid
      category="dragon ball"
      key="dragon ball"
    />
  </ol>
</Fragment>
`;
